demos/super_resolution_demo/main.cpp

```diff
@@ -92,44 +92,62 @@ int main(int argc, char *argv[]) {
         slog::info << "Preparing input blobs" << slog::endl;
 
         /** Taking information about all topology inputs **/
-        InputsDataMap inputInfo(network.getInputsInfo());
+        ICNNNetwork::InputShapes inputShapes(network.getInputShapes());
 
-        if (inputInfo.size() != 1 && inputInfo.size() != 2)
+        if (inputShapes.size() != 1 && inputShapes.size() != 2)
             throw std::logic_error("The demo supports topologies with 1 or 2 inputs only");
-
-        const std::string lrInputBlobName = "0";
+        std::string lrInputBlobName = inputShapes.begin()->first;
+        const SizeVector &lrShape = inputShapes[lrInputBlobName];
+        // A model like single-image-super-resolution-???? may take bicubic interpolation of the input image as the
+        // second input
+        std::string bicInputBlobName;
+        if (inputShapes.size() == 2) {
+            bicInputBlobName = (++inputShapes.begin())->first;
+            const SizeVector &bicShape = inputShapes[bicInputBlobName];
+            if (lrShape.size() != 4 || bicShape.size() != 4) {
+                throw std::logic_error("Number of dimensions for both inputs must be 4");
+            }
+            if (lrShape[2] >= bicShape[2] && lrShape[3] >= bicShape[3]) {
+                lrInputBlobName.swap(bicInputBlobName);
```

Comment:
You should also swap `lrShape` and `bicShape`, because `lrShape` is used below and it needs to correspond to `lrInputBlobName`.